### PR TITLE
Use REST API for the disk selection in the select_disk scenario

### PR DESCRIPTION
**Pull request by rwx788**
In guided setup we have additional steps in case multiple disks are
available, so user can select which disks to use, select which disk use
for the root partition and what to do with the existing partitions.

Adding initial structure to handle this step of guided setup.

Once we finish migrating Guided Partitioning to REST API, we can re-use it in all tests.

See [poo#89932](https://progress.opensuse.org/issues/89932).

## Verification runs
* [SLES](https://openqa.suse.de/tests/overview?distri=sle&version=15-SP3&build=rwx788%2Fos-autoinst-distri-opensuse%23yast)
* [openSUSE](https://openqa.opensuse.org/tests/1684869)

### CODE REVIEW

File: lib/Installation/Partitioner/LibstorageNG/v4_3/GuidedSetupController.pm [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+=cut
+
+sub guided_setup {
+    my ($self, %args) = @_;
+    $self->get_suggested_partitioning_page()->press_guided_setup_button();
+    if (my @disks = @{$args{disks}}) {
+        $self->get_select_disks_to_use_page()->select_hard_disks(@disks);
+    }
+    if ($args{existing_partitions}) {
+        $self->get_select_disks_partitions_page()->skip_handling_partitions();
```
Comment (jknphy) [inline lib/Installation/Partitioner/LibstorageNG/v4_3/GuidedSetupController.pm:78]: what about next if the key is not defined putting the is_shown every time we get the page? We could just have some setup method which is partially implemented which does next at the end, and when we need more we just put there some code. Do you like more to use skip explicitly for some reason, I'm facing similar things, so I'm not sure. What I remember we agreed a time ago was to have different methods if there is different flow of screens but it is not this case.

Comment (jknphy) [reply lib/Installation/Partitioner/LibstorageNG/v4_3/GuidedSetupController.pm:78]: in fact, at least the first one is the case, where the screen is optional depending of the number or disk, the other one I don't know, I didn't see it in the verifications, so perhaps two different methods make it more clear, but there is not big different though.

File: lib/Installation/Partitioner/LibstorageNG/GuidedSetupController.pm [hunk 2]
```diff
@@ -25,14 +25,20 @@ use Installation::Partitioner::LibstorageNG::SelectHardDisksPage;
...
     my ($class, $args) = @_;
-    my $self = bless {
-        SuggestedPartitioningPage => Installation::Partitioner::LibstorageNG::SuggestedPartitioningPage->new(),
-        PartitioningSchemePage    => Installation::Partitioner::LibstorageNG::PartitioningSchemePage->new(),
-        TooSimplePasswordDialog   => Installation::Partitioner::LibstorageNG::TooSimplePasswordDialog->new(),
-        FileSystemOptionsPage     => Installation::Partitioner::LibstorageNG::FileSystemOptionsPage->new(),
-        FileSystemOptionsLvmPage  => Installation::Partitioner::LibstorageNG::FileSystemOptionsLvmPage->new(),
-        SelectHardDisksPage       => Installation::Partitioner::LibstorageNG::SelectHardDisksPage->new()
-    }, $class;
+    my $self = bless {}, $class;
```
[the remark this answers is not in the document]
Comment (rwx788) [reply lib/Installation/Partitioner/LibstorageNG/GuidedSetupController.pm:28]: What do you mean exactly?

File: lib/Installation/Partitioner/LibstorageNG/v4_3/GuidedSetupController.pm [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    my ($self, $args) = @_;
+    $self->SUPER::init($args);
+    $self->{SelectDisksToUsePage}      = Installation::Partitioner::LibstorageNG::v4_3::SelectDisksToUsePage->new({app => YuiRestClient::get_app()});
+    $self->{SelectDisksPartitionsPage} = Installation::Partitioner::LibstorageNG::v4_3::SelectDisksPartitionsPage->new({app => YuiRestClient::get_app()});
+    return $self;
+}
+
+sub get_select_disks_partitions_page {
+    my ($self) = @_;
+    return $self->{SelectDisksPartitionsPage};
```
[the remark this answers is not in the document]
Comment (rwx788) [reply lib/Installation/Partitioner/LibstorageNG/v4_3/GuidedSetupController.pm:51]: Ah, right, it's actually better place to put `is_shown` check. Will move it there.

File: lib/Installation/Partitioner/LibstorageNG/v4_3/GuidedSetupController.pm (hunk 1 above)
Comment (rwx788) [reply lib/Installation/Partitioner/LibstorageNG/v4_3/GuidedSetupController.pm:78]: This is the way we have it currently in older guided setup, in general we don't do anything on the screen with existing partitions. I've realized that I didn't have to implement this either, as `select_disk` is not executed anywhere where we expect existing partitions. Which is not the case in lvm tests where we boot into image created by `cryptlvm` scenario. There we need this functionality. As for naming, I will spend more time on it, that's why PR is WIP. But `skip_handling_partitions` is the name we already have in the previous implementation.

File: lib/Installation/Partitioner/LibstorageNG/GuidedSetupController.pm (hunk 2 above)
[the remark this answers is not in the document]
Comment (jknphy) [reply lib/Installation/Partitioner/LibstorageNG/GuidedSetupController.pm:28]: year are maintainer as usual :)

File: tests/installation/partitioning/guided_setup_multiple_disks.pm [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: The test module performs guided partitioning
```
Comment (jknphy) [inline tests/installation/partitioning/guided_setup_multiple_disks.pm:10]: we could also just name it 'guided_setup.pm' for now.

Comment (rwx788) [reply tests/installation/partitioning/guided_setup_multiple_disks.pm:10]: Yes, that's what I wanted to ask you, if we are aiming data driven test module, or I should introduce the method which aims multidisk installations only. So yes, let's call it `guided_setup`.

File: lib/Installation/Partitioner/LibstorageNG/GuidedSetupController.pm (hunk 2 above)
[the remark this answers is not in the document]
Comment (rwx788) [reply lib/Installation/Partitioner/LibstorageNG/GuidedSetupController.pm:28]: Ah, sure ;)

**Comment by jknphy** [comment]
LGTM

**Comment by dzedro** [comment]
This broke https://openqa.suse.de/tests/5746650#step/no_separate_home/2

**Pull request merged**